test/search_index_ddoc_update_tests.erl

```diff
@@ -0,0 +1,277 @@
+        {foreachx, fun setup/1, fun teardown/2, [{Mod, Fun} || Fun <- Funs]}
+    }.
+
+should_stop_indexing_after_deleting_index(PortType, {Host, DbName}) ->
+    {timeout, 60,
+     ?_test(begin
+         ok = meck:reset(clouseau_rpc),
+         ok = meck:reset(dreyfus_index),
+         ok = create_doc(PortType, ?l2b(DbName), <<"doc_id">>, {[]}),
+         ReqUrl = Host ++ "/" ++ DbName ++ "/_design/searchddoc/_search/other?q=*:*",
+         {ok, Status, _Headers, _Body} =
+              test_request:get(ReqUrl, [?AUTH]),
+          ?assertEqual(200, Status),
+          DDocId = "_design/searchddoc",
+          Sig = ensure_index_is_opened(DbName, DDocId, "other"),
+
+          % Upload the new DDoc which will delete the search index named "other". And verify that the index will be closed
+          UpdatedDDocJsonObj = updated_ddoc_json_obj(DbName, ?l2b(DDocId)),
+          upload_ddoc(Host, DbName, UpdatedDDocJsonObj),
+          ensure_index_is_closed(DbName, "_design/searchddoc", "other", Sig)
+    end)}.
+
+should_stop_indexing_after_deleting_ddoc(PortType, {Host, DbName}) ->
+    {timeout, 60,
+     ?_test(begin
+         ok = meck:reset(clouseau_rpc),
+         ok = meck:reset(dreyfus_index),
+         ok = create_doc(PortType, ?l2b(DbName), <<"doc_id">>, {[]}),
+         ReqUrl = Host ++ "/" ++ DbName ++ "/_design/searchddoc/_search/other?q=*:*",
+         {ok, Status, _Headers, _Body} =
+              test_request:get(ReqUrl, [?AUTH]),
+          ?assertEqual(200, Status),
+          DDocId = "_design/searchddoc",
+          Sig = ensure_index_is_opened(DbName, DDocId, "other"),
+
+          % Delete the design doc and verify that the index will be closed
+          delete_ddoc(Host, DbName, ?l2b(DDocId)),
+          ensure_index_is_closed(DbName, "_design/searchddoc", "other", Sig)
+    end)}.
+
+updated_ddoc_json_obj(DbName, DDocId) ->
+    {ok, DDoc} = fabric:open_doc(DbName, <<DDocId/binary>>, [ejson_body, ?ADMIN_CTX]),
+    % {DDocJsonObj} = ?DDOC_UPDATE,
+    % Rev = {<<"_rev">>, couch_doc:rev_to_str(DDoc#doc.revs)},
+    % DDocUpdated = {[Rev] ++ DDocJsonObj},
+    % DDocUpdated.
+    UpdatedDDocJsonObj =
+        {[
+            {<<"_id">>, <<"_design/searchddoc">>},
+            {<<"_rev">>, ?b2l(couch_doc:rev_to_str(DDoc#doc.revs))},
+            {<<"indexes">>, {[
+                {<<"default">>, {[
+                    {<<"index">>, <<"function(doc){if(doc.blah){index(\\\"default\\\", doc.blah);}}">>}
+                ]}}
+            ]}}
+        ]},
+    UpdatedDDocJsonObj.
+
+ensure_index_is_opened(DbName, DDocId, IndexName) ->
+    IndexWithName = getIndex(DbName, ?l2b(DDocId), ?l2b(IndexName)),
+    ?assertMatch(#index{sig=_Sig}, IndexWithName),
+    [#index{sig=Sig}] = [IndexWithName],
+    IndexPidsFromEts = get_index_by_sig_from_ets(?l2b(DDocId), Sig),
+    ?assert(length(IndexPidsFromEts) > 0),
+    Sig.
+
+ensure_index_is_closed(DbName, DDocId, IndexName, Sig) ->
```

Comment:
How do you know that index Pids are actually killed? It seems like you rely on the info from ets. In this case if we would introduce a bug in a Pid accounting in ets tables we wouldn't be able to catch the problem.
You could pass the Pids of indexer instances as an argument to this function or make this check in the test case itself.